### DISCUSSION TITLE
Error when instanciating a HighsLp object

### DISCUSSION THREAD

**Discussion by leduqueadrien** [ERGO-Code/HiGHS#1569] [General]
Hello,
I am using Highs for a cpp project. In this project, I need to create a HighsLp object. Unfortunately, this error occurs :
`Exception thrown at 0x00007FFB3F2C076C (highs.dll) in TestHighsLp.exe: 0xC0000005: Access violation writing location 0x0000000000000000.`

I made a test project to isolate the issue. Here are the cpp file and the cmake configuration file :

```cpp
#include <Highs.h>
#include <iostream>

int main(int, char const *[])
{
    HighsLp m_lp;
    return 0;
}
```

```cmake
cmake_minimum_required(VERSION 3.25)
project(TestHighs LANGUAGES CXX)

find_package(Threads REQUIRED)
find_package(HIGHS REQUIRED)

add_executable(TestHighsLp mainHighsLp.cpp)
target_link_libraries(TestHighsLp PUBLIC highs::highs)

```

I am working on Windows11 with MSVC : `Microsoft (R) C/C++ Optimizing Compiler Version 19.37.32825 for x86.`

I have tried to generate the lib from the latest release (1.6) and from master (SHA 806a56aa81ab13eed79dddd7cd36b43c87845780) and in both case, I got the issue.
The Highs lib and dll have been generated with the following configuration.
![image](https://github.com/ERGO-Code/HiGHS/assets/61196643/5593f3f3-7f57-4669-adc5-073bdd884cbb)

I have instigated a little bit more the issue :
The issue is raised by HighsLp::clear() method that has been called by the HighsLp constructor.
https://github.com/ERGO-Code/HiGHS/blob/806a56aa81ab13eed79dddd7cd36b43c87845780/src/lp_data/HighsLp.h#L24

By editing this method : 
https://github.com/ERGO-Code/HiGHS/blob/806a56aa81ab13eed79dddd7cd36b43c87845780/src/lp_data/HighsLp.cpp#L150C24-L150C24
 I find that the issue comes from the initialisation of the model_name_ and objective_name variables. 
```cpp
void HighsLp::clear() {
  std::cout << "CLEAR START" << std::endl;
  this->num_col_ = 0;
  this->num_row_ = 0;

  this->col_cost_.clear();
  this->col_lower_.clear();
  this->col_upper_.clear();
  std::cout << "CLEAR col" << std::endl;
  this->row_lower_.clear();
  this->row_upper_.clear();
  std::cout << "CLEAR row" << std::endl;

  this->a_matrix_.clear();
  std::cout << "CLEAR matrix" << std::endl;

  this->sense_ = ObjSense::kMinimize;
  std::cout << "CLEAR sense" << std::endl;
  this->offset_ = 0;
  std::cout << "CLEAR offset" << std::endl;

  std::cout << &(this->model_name_) << std::endl;
  this->model_name_ = "";
  std::cout << "CLEAR model_name" << std::endl;
  this->objective_name_ = "";
  std::cout << "CLEAR objective_name" << std::endl;

  this->new_col_name_ix_ = 0;
  this->new_row_name_ix_ = 0;
  std::cout << "CLEAR ix" << std::endl;
  this->col_names_.clear();
  this->row_names_.clear();
  std::cout << "CLEAR name" << std::endl;

  this->integrality_.clear();
  std::cout << "CLEAR integrality" << std::endl;

  this->col_hash_.clear();
  this->row_hash_.clear();
  std::cout << "CLEAR hash" << std::endl;

  this->clearScale();
  std::cout << "CLEAR scale" << std::endl;
  this->is_scaled_ = false;
  this->is_moved_ = false;
  this->cost_row_location_ = -1;
  this->has_infinite_cost_ = false;
  this->mods_.clear();
  std::cout << "CLEAR END" << std::endl;
}
```

```powershell
-------------------------------------------------------------------
You may only use the C/C++ Extension for Visual Studio Code
with Visual Studio Code, Visual Studio or Visual Studio for Mac
software to help you develop and test your applications.
-------------------------------------------------------------------
Loaded 'D:\Logiciel\Librairie\Highs\Test\HighsLp\build\Debug\TestHighsLp.exe'. Symbols loaded.
Loaded 'C:\Windows\System32\ntdll.dll'. 
Loaded 'C:\Windows\System32\kernel32.dll'. 
Loaded 'C:\Windows\System32\KernelBase.dll'. 
Loaded 'C:\Windows\System32\vcruntime140_1d.dll'. 
Loaded 'C:\Windows\System32\vcruntime140d.dll'. 
Loaded 'C:\Windows\System32\msvcp140d.dll'. 
3 Loaded 'C:\Windows\System32\ucrtbased.dll'. 
2 Unloaded 'C:\Windows\System32\ucrtbased.dll'.
Loaded 'D:\Logiciel\Librairie\Highs\install\bin\highs.dll'. Module was built without symbols.
Loaded 'C:\Windows\System32\ucrtbase.dll'. 
Loaded 'C:\Windows\System32\vcruntime140.dll'. 
Loaded 'C:\Windows\System32\msvcp140.dll'. 
Loaded 'C:\Windows\System32\vcruntime140_1.dll'. 
CLEAR START
CLEAR col
CLEAR row
CLEAR matrix
CLEAR sense
CLEAR offset
000000B1036FF2D0
Exception thrown at 0x00007FFB3E1E076C (highs.dll) in TestHighsLp.exe: 0xC0000005: Access violation writing location 0x0000000000000000.
```

When running the same code from the Highs repository by editing "call_highs_from_cpp.cpp" file, it works correctly

Does someone have any ideas how to fix this issue or how to big deeper ?

**Comment by jajhall** [comment]
Strange, I'll have a look tomorrow

**Comment by jajhall** [comment]
I've no idea what might be wrong with clearing these vectors.

**Comment by leduqueadrien** [comment]
Thank you for having take time to investigate. I will continue to search on my side

**Comment by galabovaa** [comment]
Was the HiGHS build you tried in debug or release? I will try to reproduce the error on one windows machine. Thank you for flagging this up!

**Comment by leduqueadrien** [comment]
The build was in Debug config

**Comment by leduqueadrien** [comment]
My mistake, it was in Release config. Sorry

**Comment by IIIypuk09** [comment]
I have faced with the same issue and found the root of the problem. It was different compilation options used by a project and pre-build library. In my case it's highs64int options, my project had define for it, but vcpkg prebuild library not, so incompatible types were used.

**Comment by jajhall** [comment]
That's excellent, @IIIypuk09! I'll convert this issue into a discussion

**Comment by leduqueadrien** [comment]
Hello all,
Thank you @IIIypuk09 for your answer. I didn't achieve to overcome my issue with your information. Could you provide me more :
Do you recall what was the option that you refer to ?
I didn't see any mark of Highs64int build options. Did I miss it ?
What did you do in order to resolve yours ?

Note: I find an unpleasant fix. When installing the dll with a Debug configuration (`cmake --install ./build --config Debug`), It works well
